modules/src/ics02_client/msgs.rs

```diff
@@ -15,17 +24,155 @@ pub enum ClientMsg {
 }
 
 /// A type of message that triggers the creation of a new on-chain (IBC) client.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub struct MsgCreateAnyClient {
     pub client_id: ClientId,
     pub client_type: ClientType,
     pub client_state: AnyClientState,
     pub consensus_state: AnyConsensusState,
+    pub signer: AccountId,
+}
+
+impl MsgCreateAnyClient {
+    pub fn new(
+        client_id: ClientId,
+        client_type: ClientType,
+        client_state: AnyClientState,
+        consensus_state: AnyConsensusState,
+        signer: AccountId,
+    ) -> Self {
+        MsgCreateAnyClient {
+            client_id,
+            client_type,
+            client_state,
+            consensus_state,
+            signer,
+        }
+    }
+}
+
+impl Msg for MsgCreateAnyClient {
+    type ValidationError = crate::ics24_host::error::ValidationError;
+
+    fn route(&self) -> String {
+        crate::keys::ROUTER_KEY.to_string()
+    }
+
+    fn get_type(&self) -> String {
+        TYPE_MSG_CREATE_CLIENT.to_string()
+    }
+
+    fn validate_basic(&self) -> Result<(), Self::ValidationError> {
+        // Nothing to validate since all fields are validated on creation.
+        Ok(())
+    }
+
+    fn get_sign_bytes(&self) -> Vec<u8> {
+        let mut buf = Vec::new();
+        let raw_msg: MsgCreateClient = self.clone().into();
+        prost::Message::encode(&raw_msg, &mut buf).unwrap();
+        buf
+    }
+
+    fn get_signers(&self) -> Vec<AccountId> {
+        vec![self.signer]
+    }
+}
+
+impl DomainType<MsgCreateClient> for MsgCreateAnyClient {}
+
+impl TryFrom<MsgCreateClient> for MsgCreateAnyClient {
+    type Error = Error;
+
+    fn try_from(raw: MsgCreateClient) -> Result<Self, Self::Error> {
+        let raw_client_state = match raw.client_state {
+            None => Err(Kind::DecodeMessage.context(error::Kind::InvalidRawClientState)),
+            Some(raw_state) => Ok(raw_state),
+        }?;
```

Comment:
I think your approach is more explicit (I tend to prefer it), but learning from Romain's practices I notices this more compact alternative:


```suggestion
        let raw_client_state = raw
            .client_state
            .ok_or_else(|| Kind::DecodeMessage.context(error::Kind::InvalidRawClientState))?;
```